src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -7115,6 +7115,21 @@ instruct addI_rReg_rReg_mem_ndd(rRegI dst, rRegI src1, memory src2, rFlagsReg cr
   ins_pipe(ialu_reg_mem);
 %}
 
+instruct addI_rReg_mem_rReg_ndd(rRegI dst, memory src1, rRegI src2, rFlagsReg cr)
+%{
+  predicate(UseAPX);
+  match(Set dst (AddI (LoadI src1) src2));
```

Comment:
Will this not be covered by the pattern at line 7103, since ADLC automatically generated a DFA to handle both cases?

Reply:
Will run experiments to make sure that the RegRegMem pattern also applies to RegMemReg case and remove the newly added match rules if they're redundant. Will update you soon.

Reply:
Hi @vamsi-parasa, your latest patch does not address this.